v0/destinations/hs/HSTransform-v1.js

```diff
@@ -0,0 +1,285 @@
+const get = require("get-value");
+const _ = require("lodash");
+const {
+  MappedToDestinationKey,
+  GENERIC_TRUE_VALUES
+} = require("../../../constants");
+const {
+  defaultGetRequestConfig,
+  defaultPostRequestConfig,
+  defaultRequestConfig,
+  getFieldValueFromMessage,
+  getSuccessRespEvents,
+  CustomError,
+  addExternalIdToTraits,
+  defaultBatchRequestConfig,
+  removeUndefinedAndNullValues,
+  getDestinationExternalID,
+  getDestinationExternalIDInfoForRetl
+} = require("../../util");
+const {
+  BATCH_CONTACT_ENDPOINT,
+  MAX_BATCH_SIZE,
+  TRACK_ENDPOINT,
+  IDENTIFY_CREATE_UPDATE_CONTACT,
+  IDENTIFY_CREATE_NEW_CONTACT,
+  CRM_CREATE_CUSTOM_OBJECTS
+} = require("./config");
+const {
+  getTransformedJSON,
+  getEmailAndUpdatedProps,
+  formatPropertyValueForIdentify
+} = require("./util");
+
+/**
+ * using legacy API
+ * Reference:
+ * https://legacydocs.hubspot.com/docs/methods/contacts/create_contact
+ * https://legacydocs.hubspot.com/docs/methods/contacts/create_or_update
+ *
+ * for rETL support for custom objects
+ * Ref - https://developers.hubspot.com/docs/api/crm/crm-custom-objects
+ * @param {*} message
+ * @param {*} destination
+ * @param {*} propertyMap
+ * @returns
+ */
+const processLegacyIdentify = async (message, destination, propertyMap) => {
+  const { Config } = destination;
+  const traits = getFieldValueFromMessage(message, "traits");
+  const mappedToDestination = get(message, MappedToDestinationKey);
+  // if mappedToDestination is set true, then add externalId to traits
+  // rETL source
+  if (GENERIC_TRUE_VALUES.includes(mappedToDestination?.toString())) {
```

Comment:
```
(mappedToDestination && GENERIC_TRUE_VALUES.includes(mappedToDestination.toString())) 
```

[Minor] lets do this

Reply:
The reason it was done that way is we need to check for all the cases and for `mappedToDestination: "false"`. The given suggestion will make the condition as true and break the flow

Reply:
Done